### ISSUE TITLE
TypeLoadException: Inheritance security rules violated by type: 'System.Net.Http.WebRequestHandler'. Derived types must either match the security accessibility of the base type or be less accessible.

### ISSUE THREAD

**Issue by Manojb86** [digipost/signature-api-client-dotnet#97]
Hi,

Im getting following image showing error when trying to create DirectClient object with passing client-configurations to the constructor. The creation of the client configuration done according to the documentation of the library. Used this constructor when creating client configuration.
`ClientConfiguration(Environment environment, X509Certificate2 certificate, Sender globalSender = null);`

![image](https://user-images.githubusercontent.com/6886753/27793732-dfd0b26e-601c-11e7-9fcb-cf8ba4505a36.png)

This error throwing from `Digipost.Signature.Api.Client.Core.BaseClient.MutualTlsClient()`

What is the solution for this error?

**Comment by asjafjell** [comment]
Greetings! This seems to be client side problem. A quick google search gave me the following post on [Stack Overflow](https://github.com/dotnet/corefx/issues/11100), which leads me to the following question: What version of `Sytem.Net.Http` do you have? If it is 4.1.1 or similar, then there is an incompatibility issue when paired with .NET 4.6.1.

**Comment by Manojb86** [comment]
@asjafjell The `Sytem.Net.Http` version is 4.0.0.  After going through the github issue link you provided, I upgraded `Sytem.Net.Http` version to 4.3.1 and 4.3.2.  After that i got the following error with both of newer versions.

![image](https://user-images.githubusercontent.com/6886753/27827117-5e78ab6a-60d3-11e7-8b61-f39b90f39647.png)

**Comment by asjafjell** [comment]
Great to see that you've made progress. I strongly suggest that you use [the client library documentation](http://digipost.github.io/signature-api-client-dotnet/v3.x/#install-business-certificate-in-certificate-store) for troubleshooting problems and let us know if you encounter any problems where solution is not mentioned there. 

I understand that it may be a bit tricky to find the docs when the Readme on repo root is in norwegian. This is something we will fix promptly.

**Comment by asjafjell** [comment]
Closing this after months of inactivity.

**Comment by omarpiani** [comment]
I confirm that for me and for https://github.com/dotnet/corefx/issues/11100#issuecomment-301601156 is resolved with 4.3.3
